File: app/src/main/java/com/huhx0015/androidbooster/injections/modules/ViewModelModule.java
package com.huhx0015.androidbooster.injections.modules;

import androidx.annotation.NonNull;
import androidx.appcompat.app.AppCompatActivity;
import androidx.lifecycle.ViewModelProviders;
import com.huhx0015.androidbooster.viewmodel.ApiRecyclerViewModel;
import dagger.Module;
import dagger.Provides;

@Module
public class ViewModelModule {

    /** MODULE METHODS _________________________________________________________________________ **/

    @Provides
    ApiRecyclerViewModel providesApiRecyclerViewModel(@NonNull AppCompatActivity activity) {
        return ViewModelProviders.of(activity).get(ApiRecyclerViewModel.class);
    }
}

File: app/src/main/java/com/huhx0015/androidbooster/injections/modules/ViewModelModule.kt
package com.huhx0015.androidbooster.injections.modules

import androidx.databinding.PropertyChangeRegistry
import dagger.Module
import dagger.Provides
import dagger.hilt.InstallIn
import dagger.hilt.android.components.ViewModelComponent
import dagger.hilt.android.scopes.ViewModelScoped
import io.reactivex.rxjava3.disposables.CompositeDisposable

@Module
@InstallIn(ViewModelComponent::class)
object ViewModelModule {

    @Provides
    @ViewModelScoped
    fun providesDisposable(): CompositeDisposable {
        return CompositeDisposable()
    }

    @Provides
    @ViewModelScoped
    fun providesPropertyChangeRegistry(): PropertyChangeRegistry {
        return PropertyChangeRegistry()
    }
}